File: backend/src/main/kotlin/niconicotagger/dto/api/response/SongsByVocaDbEventTagResponse.kt
package niconicotagger.dto.api.response

import niconicotagger.dto.api.misc.SongEntryByVocaDbTagForEvent
import niconicotagger.dto.inner.misc.SongType

data class SongsByVocaDbEventTagResponse(
    override val items: List<SongEntryByVocaDbTagForEvent>,
    val songTypeStats: Map<SongType, Int>,
    override val totalCount: Long
) : ApiResponse<SongEntryByVocaDbTagForEvent>

File: frontend/src/main/typescript/backend/dto/response/songsByVocaDbEventTagResponse.ts
import { SongTypeStats } from "@/backend/dto/songTypeStats";
import { SongType } from "@/backend/dto/enumeration";
import { ReleaseEvent, UpdateErrorReport } from "@/backend/dto/lowerLevelStruct";
import { DispositionRelativelyToDate } from "@/backend/dto/dateDisposition";

export interface SongsByVocaDbEventTagResponse {
  items: SongEntryByVocaDbTagForEvent[];
  songTypeStats: SongTypeStats;
  totalCount: number;
}

export interface SongEntryByVocaDbTagForEvent {
  id: number;
  name: string;
  type: string;
  artistString: string;
  publishedOn: string;
  events: ReleaseEvent[];
  disposition: DispositionRelativelyToDate;
  checked: boolean;
  toAddEvent: boolean;
  processed: boolean;
  visible: boolean;
  errorReport: UpdateErrorReport | null;
}
